src/app/header/components/TermButtons.tsx

```diff
@@ -0,0 +1,67 @@
+import { Button, ButtonGroup } from '@chakra-ui/react';
+import React, { useCallback, useContext, useMemo } from 'react';
+
+import { useTerm } from '../../context/TermContext';
+import { getButtonTerms, getTerms } from '../shared/utils';
+
+export function TermButtons() {
+  const [term, setTerm] = useTerm();
+  const status = useMemo(
+    () => ({
+      first: true,
+      second: false,
+      third: false,
+    }),
+    []
+  );
+
+  const termNames = getButtonTerms();
+  const terms = getTerms();
+
+  const onClick = useCallback(
+    (event: React.MouseEvent<HTMLElement, MouseEvent>, name: string) => {
+      event.preventDefault();
+      setTerm(name);
+      if (name === terms[0] && !status.first) {
+        status.first = true;
+        status.second = false;
+        status.third = false;
```

Comment:
There _has_ to be a way to set every key value to a certain value and change the one to true...